src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/GovNotifyEmailService.kt

```diff
@@ -20,27 +20,10 @@ class GovNotifyEmailService(
   }
 
   override fun send(email: Email): Result<Pair<UUID, TemplateId>> =
-    when (email) {
-      is NewCourtBookingUserEmail -> send(email, emailTemplates.newCourtBookingUser)
-      is NewCourtBookingPrisonCourtEmail -> send(email, emailTemplates.newCourtBookingPrisonCourtEmail)
-      is NewCourtBookingPrisonNoCourtEmail -> send(email, emailTemplates.newCourtBookingPrisonNoCourtEmail)
-      is AmendedCourtBookingUserEmail -> send(email, emailTemplates.amendedCourtBookingUser)
-      is AmendedCourtBookingPrisonCourtEmail -> send(email, emailTemplates.amendedCourtBookingPrisonCourtEmail)
-      is AmendedCourtBookingPrisonNoCourtEmail -> send(email, emailTemplates.amendedCourtBookingPrisonNoCourtEmail)
-      is CancelledCourtBookingUserEmail -> send(email, emailTemplates.cancelledCourtBookingUser)
-      is CancelledCourtBookingPrisonCourtEmail -> send(email, emailTemplates.cancelledCourtBookingPrisonCourtEmail)
-      is CancelledCourtBookingPrisonNoCourtEmail -> send(email, emailTemplates.cancelledCourtBookingPrisonNoCourtEmail)
-      is CourtBookingRequestUserEmail -> send(email, emailTemplates.courtBookingRequestUser)
-      is CourtBookingRequestPrisonCourtEmail -> send(email, emailTemplates.courtBookingRequestPrisonCourtEmail)
-      is CourtBookingRequestPrisonNoCourtEmail -> send(email, emailTemplates.courtBookingRequestPrisonNoCourtEmail)
-      is ProbationBookingRequestUserEmail -> send(email, emailTemplates.probationBookingRequestUser)
-      is ProbationBookingRequestPrisonProbationTeamEmail -> send(email, emailTemplates.probationBookingRequestPrisonProbationTeamEmail)
-      is ProbationBookingRequestPrisonNoProbationTeamEmail -> send(email, emailTemplates.probationBookingRequestPrisonNoProbationTeamEmail)
-      else -> throw RuntimeException("Unsupported email type ${email.javaClass.simpleName}.")
-    }
-
-  private fun send(email: Email, templateId: TemplateId) =
     runCatching {
+      val templateId = emailTemplates.templateFor(email::class.java)
+        ?: throw RuntimeException("EMAIL: Missing template ID for email type ${email.javaClass.simpleName}.")
```

Comment:
Strictly speaking this should never happen now we check in the EmailTemplates on creation, but just in case something slips through ...